client/python/lib/ovmsclient/tfs_compat/grpc/tensors.py

```diff
@@ -63,43 +96,72 @@ def make_tensor_proto(values, dtype=None, shape=None):
         >>> tensor_proto = make_tensor_proto(data)
         >>> print(tensor_proto)
     '''
-
-    raise NotImplementedError
-
-def _as_numpy_dtype(tensor_dtype):
-    if tensor_dtype == DataType.DT_INVALID:
-        raise ValueError("Tensor data type invalid")
-    elif tensor_dtype == DataType.DT_FLOAT:
-        return np.float32
-    elif tensor_dtype == DataType.DT_DOUBLE:
-        return np.float64
-    elif tensor_dtype == DataType.DT_INT32:
-        return np.int32
-    elif tensor_dtype == DataType.DT_UINT8:
-        return np.uint8
-    elif tensor_dtype == DataType.DT_INT16:
-        return np.int16
-    elif tensor_dtype == DataType.DT_INT8:
-        return np.int8
-    elif tensor_dtype == DataType.DT_STRING:
-        return np.str
-    elif tensor_dtype == DataType.DT_COMPLEX64:
-        return np.complex64
-    elif tensor_dtype == DataType.DT_COMPLEX128:
-        return np.complex128
-    elif tensor_dtype == DataType.DT_INT64:
-        return np.int64
-    elif tensor_dtype == DataType.DT_BOOL:
-        return np.bool
-    elif tensor_dtype == DataType.DT_UINT16:
-        return np.uint16
-    elif tensor_dtype == DataType.DT_UINT32:
-        return np.uint32
-    elif tensor_dtype == DataType.DT_UINT64:
-        return np.uint64
+    if shape is None:
+        inferred_shape = []
+    elif _is_shape_valid(shape):
+        inferred_shape = list(shape)
     else:
-        return np.uint32
-
+        raise TypeError(f'shape type should be list or tuple, but is {type(shape).__name__}')
+    
+    tensor_values = values
+    # scalars are packed into 1 elemt list
+    if np.isscalar(tensor_values):
+        tensor_values = [tensor_values]
+
+    # create np.ndarray from values and find its dtype if not provided
+    if _is_array_like(tensor_values):
+        if dtype is None:
+            tensor_values = np.array(tensor_values)
+            tensor_type = NP_TO_TENSOR_MAP.get(tensor_values.dtype.type)
+            if tensor_type is not None:
+                dtype = tensor_type.TensorDtype
+            elif tensor_values.dtype.type == np.bytes_:
+                dtype = DT_STRING
+            else:
+                raise TypeError(f"values provided are not valid")
+        # binary input
+        elif dtype == DT_STRING:
+            tensor_values = np.array(tensor_values)
+        else:
+            np_dtype = TENSOR_TO_NP_MAP.get(dtype)
+            if np_dtype is None:
+                raise TypeError(f"{dtype} is not valid dtype value")
+            
+            tensor_values = np.array(tensor_values)
+            # input was binary, but dtype was not DT_STRING
+            if tensor_values.dtype.type == np.bytes_:
+                tensor_values = np.frombuffer(tensor_values.tobytes(), dtype=np_dtype)
+            else:
+                tensor_values = tensor_values.astype(np_dtype)
```

Comment:
You mentioned that this part may generate exceptions. Can end user get some useful information from them or should we throw our own here?